libs/wingsdk/src/target-tf-aws/api.ts

```diff
@@ -0,0 +1,157 @@
+      join(__dirname, "api.ongetrequest.inflight.js"),
+      "ApiOnGetRequestHandlerClient"
+    );
+    const fn = new cloud.Function(
+      this.node.scope!, // ok since we're not a tree root
+      `${this.node.id}-OnGetRequest-${hash}`,
+      functionHandler,
+      props
+    );
+    if (!(fn instanceof Function)) {
+      throw new Error("Queue only supports creating tfaws.Function right now");
+    }
+    const _tf_function: Function = fn;
+
+    new ApiGatewayIntegration(this, `integration-${hash}`, {
+      restApiId: this.api.id,
+      resourceId: apiResource.id,
+      httpMethod: apiMethod.httpMethod,
+      integrationHttpMethod: "POST",
+      type: "AWS_PROXY",
+      uri: _tf_function.invokeArn,
+    });
+
+    const region = new DataAwsRegion(this, "region");
+    const account = new DataAwsCallerIdentity(this, "account");
+
+    new LambdaPermission(this, `permission-${hash}`, {
+      statementId: "AllowExecutionFromAPIGateway",
+      action: "lambda:InvokeFunction",
+      functionName: _tf_function._functionName,
+      principal: "apigateway.amazonaws.com",
+      sourceArn: `arn:aws:execute-api:${region.name}:${account.accountId}:${this.api.id}/*/${apiMethod.httpMethod}${apiResource.path}`,
+    });
+
+    new LambdaPermission(this, "apigw-lambda", {
+      functionName: _tf_function._functionName,
+      action: "lambda:InvokeFunction",
+      principal: "apigateway.amazonaws.com",
+      sourceArn: `${this.api.executionArn}/*/*`,
+    });
+
+    const deployment = new ApiGatewayDeployment(this, "deployment", {
+      restApiId: this.api.id,
+      lifecycle: {
+        createBeforeDestroy: true,
+      },
+    });
+    const stage = new ApiGatewayStage(this, "stage", {
+      restApiId: this.api.id,
+      stageName: "example",
+      deploymentId: deployment.id,
+    });
+    new ApiGatewayMethodSettings(this, "method-settings", {
+      restApiId: this.api.id,
+      stageName: stage.stageName,
+      settings: {},
+      methodPath: "*/*",
+    });
+
+    core.Resource.addConnection({
+      from: this,
+      to: fn,
+      relationship: "on_get_request",
+    });
+  }
+
+  /** @internal */
+  public _bind(host: core.IInflightHost, ops: string[]): void {
+    if (!(host instanceof Function)) {
+      throw new Error("queues can only be bound by tfaws.Function for now");
+    }
+
+    super._bind(host, ops);
+  }
+
+  /** @internal */
+  public _toInflight(): Code {
+    throw new Error("Method not implemented.");
+  }
+}
+
+/**
+ * Represents a resource with an inflight "handle" method that can be passed to
+ * `Api.get`.
+ *
+ * @inflight `wingsdk.cloud.IApiOnGetRequestHandlerClient`
+ */
+export interface IApiOnGetRequestHandler extends IResource {}
+
+/**
+ * Inflight client for `IGetRequestHandler`.
+ */
+export interface IApiOnGetHandlerHandlerClient {
+  /**
+   * Function that will be called when a get request is made to the url.
+   * @inflight
+   */
+  handle(message: string): Promise<void>;
+}
```

Comment:
This should be in `cloud/api.ts` (the inflight interface should be the same among all clouds)